### ISSUE TITLE
Repeated extra arguments not respected

### ISSUE THREAD

**Issue by terrycain** [k3s-io/k3s#5374]
**Environmental Info:**
K3s Version: 
```
k3s version v1.23.1+k3s1 (08d538fb)
go version go1.17.5
```

Node(s) CPU architecture, OS, and Version: 
```
Linux kube01 5.4.0-107-generic #121-Ubuntu SMP Thu Mar 24 16:04:27 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
```

Cluster Configuration:
```
1 server
```

**Describe the bug:**
When providing repeated extra arguments to the API server, they are not passed to the API server in the expected manner
```
--kube-apiserver-arg='service-account-issuer=https://s3-eu-west-2.amazonaws.com/somebucket' \
--kube-apiserver-arg='service-account-issuer=https://kubernetes.default.svc.cluster.local' \
```
This results in only `--service-account-issuer='https://kubernetes.default.svc.cluster.local'` being passed to the API server.

**Steps To Reproduce:**

Run 
```
/usr/local/bin/k3s server \
  --kube-apiserver-arg='service-account-issuer=https://s3-eu-west-2.amazonaws.com/somebucket' \
  --kube-apiserver-arg='service-account-issuer=https://kubernetes.default.svc.cluster.local'
```
Then verify the output where it says `Running kube-apiserver FLAGS....`

**Expected behavior:**

When `k3s` logs out the API server arguments, it lists the repeated arguments, in the order they are specified (as that matters for `--service-account-issuer`)

**Actual behavior:**

Only the last repeated argument is passed to the API server.

**Backporting**

- [ ] Needs backporting to older releases

Not sure if it should be backported.

PR here - https://github.com/k3s-io/k3s/pull/5373

--- 
Extra info

If you try and specify the service-account-issuer with a comma separated string you get an invalid issuer in the eks service account token

Args:
```
--kube-apiserver-arg='service-account-issuer=https://s3-eu-west-2.amazonaws.com/oidc-test-wbdkdthrsxepwhsbvrjdgpcjrlokgrcr,https://kubernetes.default.svc.cluster.local' 
--kube-apiserver-arg='api-audiences=https://kubernetes.default.svc.cluster.local,k3s,aws-sts' 
```

Token:
```
eyJhbGciOiJSUzI1NiIsImtpZCI6IlAyVTVhVHJ5MFBwSVdSVHdsRzgtY2czdXpnaHhfNGNSRVhXandrR19jXzAifQ.eyJhdWQiOlsiYXdzLXN0cyJdLCJleHAiOjE2NDg5NDEzMTIsImlhdCI6MTY0ODg1NDkxMiwiaXNzIjoiaHR0cHM6Ly9zMy1ldS13ZXN0LTIuYW1hem9uYXdzLmNvbS9vaWRjLXRlc3Qtd2Jka2R0aHJzeGVwd2hzYnZyamRncGNqcmxva2dyY3IsaHR0cHM6Ly9rdWJlcm5ldGVzLmRlZmF1bHQuc3ZjLmNsdXN0ZXIubG9jYWwiLCJrdWJlcm5ldGVzLmlvIjp7Im5hbWVzcGFjZSI6ImRlZmF1bHQiLCJwb2QiOnsibmFtZSI6ImEtODU3YmY2NDk2Ny02cXB3ZiIsInVpZCI6IjBhM2VkYTEyLTEyYWYtNGJjYi1hYWFkLWVjZTcxMDc0Njg2MCJ9LCJzZXJ2aWNlYWNjb3VudCI6eyJuYW1lIjoidGVzdCIsInVpZCI6IjJlZmViZjlmLTQwOTgtNDE0Ni1iYjAxLWY4ZjM2NzAwODQ5ZCJ9fSwibmJmIjoxNjQ4ODU0OTEyLCJzdWIiOiJzeXN0ZW06c2VydmljZWFjY291bnQ6ZGVmYXVsdDp0ZXN0In0.VgCva-6hty8jGq4diD7YxyDjUA_sHCl7i7Fu6wgrUEHfUd1tq-ZzuaIPN6D5cSz3ItCc8Id4Ff0n6RC2Y-LWEzyg-VH7jGn1KVo7wrmjy9KMekdmC6qf8nXUTteg35PvLG1ScUyvmjeTV4WjVSxpFv93c3K4w_uDsZ67vJ_JJB7E_9bPggSo4x94ChG6GQiz6Rysqa6uK_EIHwzjA6-ZYtOu46YP_9HBhjal2T2-XjSxeIHxDaAjRdwjUjLkMxCNGMiRUiVnkao0i_qO80OgRn-QGHOgKm7mYE2eLe1UtXCbjXxsxL3QkHtSc08CuX1ZH8-EAqG2gfMBL4GNsGyYsw
```

**Comment by mdrahman-suse** [comment]
**Reproduced the issue in** [v1.23.5+k3s1](https://github.com/k3s-io/k3s/releases/tag/v1.23.5%2Bk3s1)

```
# Version
k3s -v
k3s version v1.23.5+k3s1 (313aaca5)
go version go1.17.5
```

- Reproduction steps
```
# Install k3s

curl -sfL https://get.k3s.io | INSTALL_K3S_EXEC="--write-kubeconfig-mode 644 --token secret --cluster-init" INSTALL_K3S_SKIP_START=true INSTALL_K3S_VERSION=v1.23.5+k3s1 sh -
```
```
# Start k3s with extra args as mentioned in the ticket

sudo k3s server \
  --kube-apiserver-arg='service-account-issuer=https://s3-us-east-2.amazonaws.com/test-bucket' \
  --kube-apiserver-arg='service-account-issuer=https://kubernetes.default.svc.cluster.local'
```
```
# From log
...
Running kube-apiserver ...
 --service-account-issuer=https://kubernetes.default.svc.cluster.local --service-account-key-file=/var/lib/rancher/k3s/server/tls/service.key ...
...
```

**Verified the fix in master with commit** 6a8de31a

```
k3s -v
k3s version v1.23.5+k3s-6a8de31a (6a8de31a)
go version go1.17.5
```

- Verification steps
```
# Install k3s

curl -sfL https://get.k3s.io | INSTALL_K3S_EXEC="--write-kubeconfig-mode 644 --token secret --cluster-init" INSTALL_K3S_SKIP_START=true INSTALL_K3S_COMMIT=6a8de31a929ff9ad15de0e11c35ddefa574d4f6a sh -
```
```
# Start k3s with extra args as mentioned in the ticket

sudo k3s server \
  --kube-apiserver-arg='service-account-issuer=https://s3-us-east-2.amazonaws.com/test-bucket' \
  --kube-apiserver-arg='service-account-issuer=https://kubernetes.default.svc.cluster.local'
```
```
# From log

Running kube-apiserver ... --service-account-issuer=https://s3-us-east-2.amazonaws.com/test-bucket --service-account-issuer=https://kubernetes.default.svc.cluster.local --service-account-key-file=/var/lib/rancher/k3s/server/tls/service.key ...
```

**Comment by ShylajaDevadiga** [comment]
Validated on k3s version v1.23.6-rc4+k3s1 both values passed to the arg is present 
```
Running kube-apiserver --advertise-port=6443 ...
--service-account-issuer=https://s3-eu-west-2.amazonaws.com/somebucket --service-account-issuer=https://kubernetes.default.svc.cluster.local --service-account-key-file=/var/lib/rancher/k3s/server/tls/service.key
...
```
